internal/controller/providers/pxc/applier.go

```diff
@@ -300,6 +291,103 @@ func (p *applier) Monitoring() error {
 	return nil
 }
 
+func (p *applier) PodSchedulingPolicy(systemNamespace string) error {
+	// NOTE: this method shall be called after Engine() and Proxy() methods
+	// because it extends the engine and proxy specs with the affinity rules.
+	//
+	// The following cases are possible:
+	// 1. The user did not specify a .spec.podSchedulingPolicyName -> do nothing.
+	// 2. The user specified a .spec.podSchedulingPolicyName, but it does not exist -> return error.
+	// 3. The user specified a .spec.podSchedulingPolicyName, and it exists, but it is not applicable to the upstream cluster -> return error.
+	// 4. The user specified a .spec.podSchedulingPolicyName, and it exists, but affinity configuration is absent there -> do nothing.
+	// 5. The user specified a .spec.podSchedulingPolicyName, and it exists, and it is applicable to the upstream cluster ->
+	// copy the affinity rules to the upstream cluster spec from policy.
+
+	pxc := p.PerconaXtraDBCluster
+	// --------------------------------- //
+	// Special workaround, need to reset all affinity params in p.PerconaXtraDBCluster before moving further.
+	// TODO: Remove it once https://perconadev.atlassian.net/browse/EVEREST-2023 is addressed
+	pxc.Spec.PXC.PodSpec.Affinity = nil
+	if pxc.Spec.HAProxy != nil {
+		pxc.Spec.HAProxy.PodSpec.Affinity = nil
+	}
+
+	if pxc.Spec.ProxySQL != nil {
+		pxc.Spec.ProxySQL.PodSpec.Affinity = nil
+	}
+	// --------------------------------- //
+
+	pspName := p.DB.Spec.PodSchedulingPolicyName
+	if pspName == "" {
+		// Covers case 1.
+
+		// We preserve the settings for existing DBs, otherwise restarts are seen when upgrading Everest.
+		// TODO: Remove this once we figure out how to apply such spec changes without automatic restarts.
+		// See: https://perconadev.atlassian.net/browse/EVEREST-1413
+		// if p.DB.Status.Status == everestv1alpha1.AppStateReady {
+		// 	pxc.Spec.PXC.PodSpec.Affinity = p.currentPerconaXtraDBClusterSpec.PXC.Affinity
+		// }
+		// // Changing proxy type is not supported, so we copy affinity rules from the previous state of the same proxy type
+		// switch p.DB.Spec.Proxy.Type {
+		// case everestv1alpha1.ProxyTypeHAProxy:
+		// 	pxc.Spec.HAProxy.PodSpec.Affinity = p.currentPerconaXtraDBClusterSpec.HAProxy.PodSpec.Affinity
+		// case everestv1alpha1.ProxyTypeProxySQL:
+		// 	pxc.Spec.ProxySQL.PodSpec.Affinity = p.currentPerconaXtraDBClusterSpec.ProxySQL.PodSpec.Affinity
```

Comment:
Note: in theory, `p.currentPerconaXtraDBClusterSpec.HAProxy` and `p.currentPerconaXtraDBClusterSpec.ProxySQL` can be nil

Reply:
the same as for PSMDB